server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -1394,13 +1384,8 @@ private DeleteResult deleteInLucene(Delete delete, DeletionStrategy plan) throws
             return new DeleteResult(
                 plan.versionOfDeletion, getPrimaryTerm(), delete.seqNo(), plan.currentlyDeleted == false);
         } catch (Exception ex) {
-            if (indexWriter.getTragicException() == null) {
-                // there is no tragic event and such it must be a document level failure
-                return new DeleteResult(
-                        ex, plan.versionOfDeletion, delete.primaryTerm(), delete.seqNo(), plan.currentlyDeleted == false);
-            } else {
-                throw ex;
-            }
+            assert indexWriter.getTragicException() != null : "delete operation should never fail at document level";
```

Comment:
can we do this like this:

```Java

if (indexWriter.getTragicException() == null) {
  throw new AssertionError("delete operation should never fail at document level", ex);
}
```
we otherwise loose the original exception and then we are in trouble :)